denonavr/foundation.py

```diff
@@ -112,6 +124,8 @@ async def async_setup(self) -> None:
             self.api.add_appcommand_update_tag(
                 AppCommands.GetAllZonePowerStatus)
 
+            await self.telnet_api.async_connect()
```

Comment:
Telnet should be optional. You don't know if every receiver supports telnet. Additionally, telnet supports only one connection per receiver and there might be cases where multiple instances of denonavr are running in parallel.

I would prefer if we just have two functions like `async_telnet_connect` and `async_telnet_disconnect` here.

Reply:
Agreed on the optional piece. Can you explain the `I would prefer if we just have two functions like async_telnet_connect and async_telnet_disconnect here.`? Are you saying you'd rather the caller have to call async_telnet_connect themselves?